### PR TITLE
AM-1013 email verification disable confirm

### PR DESCRIPTION
**Pull request by ashraf706**
**How To Test**

Steps are written in the JIRA issue

### CODE REVIEW

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/email/impl/EmailServiceImpl.java [hunk 1]
```diff
@@ -60,7 +69,10 @@
  * @author GraviteeSource Team
  */
 public class EmailServiceImpl implements EmailService, InitializingBean {
+    private static final Logger logger = LoggerFactory.getLogger(EmailServiceImpl.class);
 
+    public static final String CLIENT_ID = "client_id";
+    public static final String REDIRECT_URI = "redirect_uri";
```
Comment (leleueri) [inline gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/email/impl/EmailServiceImpl.java:75]: **todo:** We should use the constants defined into the oauth2.Parameters class https://github.com/gravitee-io/gravitee-access-management/blob/master/gravitee-am-common/src/main/java/io/gravitee/am/common/oauth2/Parameters.java

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/email/impl/EmailServiceImpl.java [hunk 2]
```diff
@@ -232,8 +244,12 @@ private Map<String, Object> prepareEmailParams(User user, Client client, Integer
...
         }
 
-        if (client != null) {
-            queryParams.add("client_id", encodeURIComponent(client.getClientId()));
+        if (client != null && !queryParams.contains(CLIENT_ID)) {
+            queryParams.add(CLIENT_ID, encodeURIComponent(client.getClientId()));
+        }
+
+        if(queryParams.contains(REDIRECT_URI)) {
+            encodeAndReplaceRedirectURIQueryParam(queryParams);
```
Comment (leleueri) [inline gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/email/impl/EmailServiceImpl.java:252]: **thought**: I'm not sure that this is the fix we should apply :thinking: I wondering instead with we provide a redirect_uri parameter to the validationRegistration endpoint as it not used... wdyt ?

Comment (leleueri) [reply gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/email/impl/EmailServiceImpl.java:252]: I just check, and for the forgotPassword for example, we are not providing the parameters coming from the request. That make sens IMO, I wondering why it has been done in another way for registrationValidation... I think we should do the same as forgotPassword. wdyt ?

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/email/impl/EmailServiceImpl.java (hunk 1 above)
Comment (ashraf706) [reply gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/email/impl/EmailServiceImpl.java:75]: done

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/email/impl/EmailServiceImpl.java (hunk 2 above)
Comment (ashraf706) [reply gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/email/impl/EmailServiceImpl.java:252]: code updated

**Review by leleueri — APPROVED** [review]

**Pull request merged**